server/src/test/java/io/crate/integrationtests/AggregateExpressionIntegrationTest.java

```diff
@@ -305,4 +308,33 @@ public void test_assure_cmp_by_function_call_with_reference_and_literal_does_not
         execute("select max_by(name, 1) from tbl;");
         assertThat(response).hasRows("foo");
     }
+
+    @SuppressWarnings("unchecked")
+    public void test_topk_agg() {
+        execute("create table tbl (l long, l_no_doc_values long storage with(columnstore = false))");
+        execute("insert into tbl(l, l_no_doc_values) values (1, 1), (1, 1), (1, 2), (2, 2), (2, 2)");
+        execute("refresh table tbl");
+
+        // TODO: Check why the item and frequency numbers switch between Integer & Long
```

Comment:
I still haven't found why the resulting json (map) switches between Integer and Long for the value types of item and frequence. Will continue investigating and address it separately.